### PR TITLE
Fix trailing comma for function with one arg (#880)

### PR DESCRIPTION
**Pull request by dylanjblack**
Modified maybe_remove_trailing_comma to remove trailing commas for typedarglists (in addition to arglists), and updated line split logic to ensure that all lines in a function definition that contain only one arg have a trailing comma.

Some discussion points:

- There are likely some edge cases to consider with the use of `no_commas`. Would be great to hear any other suggestions for this test. 
- The new test data file should probably be folded into one of the existing function test files, are there any guidelines around this? 
- It looks like this will clash with Pull request #826.

### CODE REVIEW

**Comment by illfated** [comment]
(ref. issue #880 )

File: black.py [hunk 1]
```diff
@@ -2488,9 +2491,12 @@ def bracket_split_build_line(
         if leaves:
             # Since body is a new indent level, remove spurious leading whitespace.
             normalize_prefix(leaves[0], inside_brackets=True)
-            # Ensure a trailing comma for imports, but be careful not to add one after
-            # any comments.
-            if original.is_import:
+            # Ensure a trailing comma for imports and standalone function arguments, but
+            # be careful not to add one after any comments.
+            no_commas = original.is_def and not [
```
Comment (JelleZijlstra) [inline black.py]: Use `any()`

File: tests/test_black.py [hunk 2]
```diff
@@ -264,6 +264,14 @@ def test_function2(self) -> None:
         black.assert_equivalent(source, actual)
         black.assert_stable(source, actual, black.FileMode())
 
+    @patch("black.dump_to_file", dump_to_stderr)
+    def test_function3(self) -> None:
```
Comment (JelleZijlstra) [inline tests/test_black.py]: It's fine to have another test file, but could you name it something less generic, like `test_function_trailing_comma`?

File: black.py (hunk 1 above)
Comment (dylanjblack) [reply black.py]: Good call, cheers.
